tests/test_dice_pool.py

```diff
@@ -0,0 +1,44 @@
+import pytest
+
+import dice
+import dice_pool
+
+
+def test_ability_class_has_proper_fields(ability_class):
+    assert hasattr(ability_class, 'dice')
```

Comment:
I see that this is another implementation of your method `__hasrequired_fields `. If you would make it a function, or at least not superprivate method, you could write a test on that. Even 2: one using `Ability` object, and another — using stub object, implementing your desired interface.
Another thing to mention here is that you should be testing not against class itself, but its instance.